schema_based_on_diagram.sql

```diff
@@ -0,0 +1,51 @@
+CREATE TABLE patients(
+   id serial PRIMARY KEY,
+   name VARCHAR(255) NOT NULL,
+   date_of_birth DATE
+);
+
+CREATE TABLE treatments(
+   id serial PRIMARY KEY,
+   type VARCHAR(100),
+   name VARCHAR(250)
+);
+
+CREATE TABLE medical_histories(
+   id serial PRIMARY KEY,
+   patient_id INT,
+   admitted_at TIMESTAMP,
+   status VARCHAR(100),
+   CONSTRAINT fk_patient
+      FOREIGN KEY(patient_id) 
+	  REFERENCES patients(id)
+	  ON DELETE CASCADE
+);
+
+CREATE TABLE invoices(
+   id serial PRIMARY KEY,
+   medical_history_id INT,
+   total_amount REAL,
```

Comment:
[ OPTIONAL]
- [ ] The `decimal` type would be better here as specified in the table
```suggestion
   total_amount DECIMAL,
```

Reply:
Hello @christianbiring1 thank you for the remarks, I will correct and commit the changes.